File: base-backend-node/src/app/api/v1/routes/news.ts
import { Hono } from 'hono';
import { getDatabases, DATABASE_ID } from '../../../services/appwriteClient';
import { Query } from 'node-appwrite';
import { Bindings } from '../../../core/types';

export const newsRouter = new Hono<{ Bindings: Bindings }>();

newsRouter.get('/', async (c) => {
  const db = getDatabases(c.env);
  try {
    const res = await db.listDocuments(DATABASE_ID, 'news', [Query.orderDesc('publishedAt')]);
    return c.json(res.documents);
  } catch (e) {
    console.error(`Error fetching news:`, e);
    return c.json([]);
  }
});

newsRouter.get('/:slug', async (c) => {
  const db = getDatabases(c.env);
  try {
    const res = await db.listDocuments(DATABASE_ID, 'news', [Query.equal('slug', c.req.param('slug'))]);
    if (res.documents.length) {
      return c.json(res.documents[0]);
    }
    return c.json(null);
  } catch (e) {
    console.error(`Error fetching news article:`, e);
    return c.json(null);
  }
});

File: base-backend/app/api/v1/routes/news.py
from fastapi import APIRouter
from app.services.appwrite_client import get_databases, DATABASE_ID
from appwrite.query import Query

router = APIRouter()

@router.get("/")
def get_news():
    db = get_databases()
    try:
        response = db.list_documents(
            database_id=DATABASE_ID,
            collection_id="news",
            queries=[Query.order_desc("publishedAt")]
        )
        return response.get("documents", [])
    except Exception as e:
        print(f"Error fetching news: {e}")
        return []

@router.get("/{slug}")
def get_news_article(slug: str):
    db = get_databases()
    try:
        response = db.list_documents(
            database_id=DATABASE_ID,
            collection_id="news",
            queries=[Query.equal("slug", slug)]
        )
        docs = response.get("documents", [])
        if docs:
            return docs[0]
        return None
    except Exception as e:
        print(f"Error fetching news article: {e}")
        return None
